lib/streaming-pipeline.js

```diff
@@ -40,9 +40,18 @@ module.exports = class StreamingPipeline extends Writable {
         this._functionOutputs = [];
         this._finishedOutputs = 0;
         this._streamMetadata = {};
-        this.on('finish', () => {
-            logger('Ending input streams');
-            this._endInputs();
+        this.on('pipe', (src) => {
```

Comment:
the end and error events should be propagated by pipe.

Reply:
I don't see that in the [documentation](https://nodejs.org/docs/latest-v12.x/api/stream.html#stream_readable_pipe_destination_options).
The `end` _function_ of the Writable is called when the Readable ends.
Nothing about error propagation.

Reply:
you're right, I misread [this snippet](https://github.com/nodejs/node/blob/4c746a6cfda980c1cd0de6246781c0083d9e416c/lib/_stream_readable.js#L737-L755). It's watching the destination stream on pipe for errors, not the source.